### DISCUSSION TITLE
No valid values defined for var=VarDef[200.Content.Media-Type]

### DISCUSSION THREAD

**Discussion by mneiferbag** [Cornutum/tcases#212] [Q&A] [answered]
I just got started with OpenAPI and Tcases. Please be patient. I have a question about the following error message from Tcases.

`[ERROR] Failed to execute goal org.cornutum.tcases:tcases-maven-plugin:3.7.1:api (default-cli) on project jaxrs: Can't generate requested models: FunctionInputDef[POST_verb]: Can't create test cases: Can't complete tuples -- no valid values defined for var=VarDef[200.Content.Media-Type] -> [Help 1]`

It seems to me, that Tcases requires a media-type for a post response 200. Afaik this is not required by the OpenAPI 3.0 spec. Is this check configurable in Tcases? I had a look at `mvn tcases:help -Ddetail=true -Dgoal=api` but found nothing.

Thanks in advance!

**Comment by kerrykimbrough** [comment, marked as answer]
You did not attach an OpenAPI definition to reproduce this problem. However, I was able to reproduce this symptom using a response definition with an empty `content` map -- see below.  But that's a peculiar way to say "no content". A simpler way is to just omit the `content` field entirely, in which case the problem goes away. Assuming this matches your situation, that would be an easy fix. Do you agree?

```json
{
    "openapi": "3.0.0",
    "info": {
        "title": "Responses",
        "version": "0.0.0"
    },
    "paths": {
        "/responses": {
            "get": {
                "responses": {
                    "200": {
                        "description": "No content",
                        "content": {
                        }
                    }
                }
            }
        }
    }
}
```

On the one hand, dropping the `content` will cure the failure and clarify the API definition. On the other hand, I agree that an empty `content` should be a valid way to say "no content", so that's a Tcases defect that ought to be fixed.

**Reply by mneiferbag** [reply]
Thank you for your answer and my apologies for omitting the definition. You can view it online. 

[https://raw.githubusercontent.com/mneiferbag/java-jaxrs/main/src/main/resources/api_v3.yaml](https://raw.githubusercontent.com/mneiferbag/java-jaxrs/main/src/main/resources/api_v3.yaml)

You are right, I use `content: {}`. Please note that I'm no expert in API design and this is just a toy project to get started with API testing and Tcases.

I have removed all occurences of `content: {}` and I can now run `mvn tcases:api` without errors. I have checked my adjusted definition with editor.swagger.io and I see no errors or warnings. I will continue with this definition.

**Reply by kerrykimbrough** [reply]
See also issue #213
